src/entity/Living.php

```diff
@@ -568,6 +600,14 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 			if($this->doAirSupplyTick($tickDiff)){
 				$hasUpdate = true;
 			}
+
+			if(
+				$this->isSwimming() && !$this->isUnderwater() &&
```

Comment:
isUnderwater() should finally be changed to check if the bounding box is submerged.
I suggest a change in syntax as follows:
`function isUnderwater(bool $checkSubmerged = false)`
This way old behaviour will be kept by default, but also allowing checks like the ones below to be removed